src/alire/alire-origins-deployers.adb

```diff
@@ -145,4 +159,55 @@ package body Alire.Origins.Deployers is
    function Deploy (This : Deployer; Folder : String) return Outcome
    is (raise Program_Error with "should never be called for base class");
 
+   -------------------
+   -- Verify_Hashes --
+   -------------------
+
+   function Verify_Hashes (This   : Deployer'Class;
+                           Folder : String) return Outcome is
+   begin
+      if Supports_Hashing (This.Base.Kind) then
+
+         --  Emit a note if we might profit from hashes:
+         if This.Base.Data.Hashes.Is_Empty then
+            Trace.Warning ("No integrity hashes provided for "
+                           & This.Base.Image);
+            --  TODO: make this an error once all crates are updated with
+            --  their hashes.
+         else
+            Trace.Detail ("Verifying integrity...");
+         end if;
+
+         --  Compute hashes from downloaded release and verify:
+         for Index_Hash of This.Base.Data.Hashes loop
```

Comment:
So if multiple hashes are provided, all of them must be correct.

Reply:
Yes. We only support one now, but I think it may come in handy for future-proofing.